pkg/controller/cluster/operator.go

```diff
@@ -317,21 +308,8 @@ func (c *Controller) createUsers(ctx context.Context, tenant *miniov2.Tenant, te
 }
 
 func (c *Controller) createBuckets(ctx context.Context, tenant *miniov2.Tenant, tenantConfiguration map[string][]byte) error {
-	var buckets []miniov2.Bucket
-	buckets = append(buckets, tenant.Spec.Buckets...)
-	// configuration that means MinIO is running with LDAP enabled
-	// and we need to skip the default bucket creation
-	var ldapEnabled bool
-	var minioClnt *minio.Client
-	for _, env := range tenant.GetEnvVars() {
-		if env.Name == "MINIO_IDENTITY_LDAP_SERVER_ADDR" && env.Value != "" {
-			ldapEnabled = true
-			break
-		}
-	}
-
 	// Skip default bucket creation for ldap setup
-	if !ldapEnabled {
+	if !tenant.IsLDAPEnabled() && len(tenant.Spec.Buckets) > 0 {
```

Comment:
LDAP can be enabled via the configuration file as well while tenant env variables will be empty, I'll double check if we have a bug here.

Reply:
LDAP can be enabled via configuration but that is not a provisioning problem - the operator does things during provision and the only way Operator will know this information is via environment variables. 

 If customers enable LDAP after the tenant is provisioned there is no problem here and operator doesn't come into place.